File: backend/src/main/java/com/etekhno/rentacar/domain/AddressDTO.java
package com.etekhno.rentacar.domain;

import com.etekhno.rentacar.domain.inbound.AddressDTOIn;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AddressDTO extends AddressDTOIn {
    String id;
    public AddressDTO(String id, String city, String street, String state, int postalCode, String country ){
        super(city, street, state, postalCode, country);
        this.id = id;
    }
}

File: frontend/lib/domains/address_dto.dart
import '/domains/inbound/address_dto_in.dart';

class AddressDTO extends AddressDTOIn {
  String? id;

  AddressDTO(
      {this.id,
      super.city,
      super.street,
      super.state,
      super.postalCode,
      super.country});

  factory AddressDTO.fromJson(Map<String, dynamic> json) {
    return AddressDTO(
      id: json['id'] ?? '',
      city: json['city'] ?? '',
      street: json['street'] ?? '',
      state: json['state'] ?? '',
      postalCode: json['postalCode'] ?? 0,
      country: json['country'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      ...super.toJson(),
    };
  }
}
